### PR TITLE
tests/main/snap-quota-{install/journal}: fix unstable spread tests

### PR DESCRIPTION
**Pull request by Meulengracht**
**snap-quota-install**
On some versions of systemd, just creating a cgroup implies usage of memory.

**snap-quota-journal**
The test checks to quickly after doing quota changes for the services to be active again, let it retry 3 times to allow for services to come online, and also make sure the restart condition for the test journal snap is set to 'always', as if systemd restarts the journal namespace while the snap is writing to the namespace, then the service will be killed with a SIGPIPE. This is a valid exit condition for systemd, and thus the service will not be restarted as `on-failure` is the default behaviour.

### CODE REVIEW

**Review by sergiocazzolato — APPROVED** [review]
Thanks

File: tests/main/snap-quota-install/task.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ execute: |
   echo "Checking that all quotas can be listed"
   snap quotas | cat -n > quotas.txt
   MATCH "     1\s+Quota\s+Parent\s+Constraints\s+Current$" < quotas.txt
-  MATCH "     2\s+group-one\s+memory=400MB\s*$" < quotas.txt
+  MATCH "     2\s+group-one\s+memory=400MB(\s*|\s*memory=[0-9.a-zA-Z]+)\s*$" < quotas.txt
```
Comment (mardy) [inline tests/main/snap-quota-install/task.yaml:32]: Please add a comment above this line, explaining that some memory is always being used. But shouldn't we be a bit less tolerant about the last column format? If it's only a couple of pages, I guess we are talking about KB, so we could avoid specifying `a-zA-Z` but rather match the exact sequence.

**Review by mardy — APPROVED** [review]
I guess that what we want to test is the contraints set by us and not the current ones, but in case you find it reasonable, you could consider being more strict with the regexp.

File: tests/main/snap-quota-install/task.yaml (hunk 1 above)
Comment (Meulengracht) [reply tests/main/snap-quota-install/task.yaml:32]: I'd rather not be less tolerant here, as we already have tests that are very exact on this area. What amount of memory is in use is totally unrelated to the test - I'll add a comment tho (nice suggestion) explaining that we expect some memory to be in use

**Pull request merged**